lib/utils/device/live.ts

```diff
@@ -162,11 +164,31 @@ export class LivepushManager {
 					eventQueue.push(changedPath);
 					this.getDebouncedEventHandler(serviceName)();
 				};
+
+				// Prepare dockerignore data for file watcher
+				const serviceDirsByService = this.deployOpts.multiDockerignore
+					? await getServiceDirsFromComposition(
+							this.deployOpts.source,
+							this.composition,
+					  )
+					: {};
+				const { ignoreByDir, serviceDirs } = await getDockerignoreByDir(
+					this.deployOpts.source,
+					serviceDirsByService,
+				);
```

Comment:
The `getServiceDirsFromComposition` and `getDockerignoreByDir` calls are not dependent on each service, so could be moved out of the `for` loop on the grounds of performance.